astropy/nddata/nddata.py

```diff
@@ -118,7 +121,7 @@ class NDData(NDDataBase):
     # Tables. It will check if the meta is dict-like or raise an exception.
     meta = MetaData(doc=_meta_doc, copy=False)
 
-    def __init__(self, data, uncertainty=None, mask=None, wcs=None,
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, psf=None,
```

Comment:
Is it safer to put new keyword in the end in case people been passing in keywords as pos args?

Reply:
I was actually wondering if we shouldn't modify this API to make everything after `data` be a keyword-only, since there's a lot of them? So it would look like: `__init__(self, data, *, uncertainty=None, ...)`.

Reply:
> While adding this, I wondered if we shouldn't modify this API to make everything after data be keyword-only, since there's a lot of them? 

I don't think we want to do this in minor version bump; there may be issues also with the `@use_nddata` decorator but I don't recall the details of that.